samples/kvs_gstreamer_sample.cpp

```diff
@@ -696,25 +696,39 @@ int gstreamer_live_source_init(int argc, char* argv[], CustomData *data, GstElem
     // Attempt to create vtenc encoder
     encoder = gst_element_factory_make("vtenc_h264_hw", "encoder");
     if (encoder) {
-        LOG_DEBUG("Using videotestsrc")
-        source = gst_element_factory_make("videotestsrc", "source");
         vtenc = true;
+        source = gst_element_factory_make("videotestsrc", "source");
+        if (source) {
+            LOG_DEBUG("Using videotestsrc");
+        } else {
+            LOG_ERROR("Failed to create videotestsrc");
+        }
     } else {
         // Failed creating vtenc - check pi hardware encoder
         encoder = gst_element_factory_make("omxh264enc", "encoder");
         if (encoder) {
+            LOG_DEBUG("Using omxh264enc");
             isOnRpi = true;
         } else {
             // - attempt x264enc
-            encoder = gst_element_factory_make("x264enc", "encoder");
             isOnRpi = false;
+            encoder = gst_element_factory_make("x264enc", "encoder");
+            if (encoder) {
+                LOG_DEBUG("Using x264enc");
+            } else {
+                LOG_ERROR("Failed to create x264enc");
+            }
         }
         source = gst_element_factory_make("v4l2src", "source");
         if (source) {
             LOG_DEBUG("Using v4l2src");
         } else {
-            LOG_DEBUG("Using ksvideosrc");
             source = gst_element_factory_make("ksvideosrc", "source");
```

Comment:
Since this is a sample it would be better if we could communicate exactly which one.  So how about a LOG line stating unable to create `v4l2src` and then the last one can only say it was unable to create `ksvideosrc` instead of combining them.

in the if(vtenc) block if we fail to create `videotestsrc` that is a fatal error we can terminate the application there.  in the else block we attempt a few so I would check after the else {} section is complete to see if we have a valid src if not, terminate the application.  I'm not a fan of having an if (!a || !b || !c ||....) thing and the customer now has to figure out well which one was it that failed.

Reply:
Sure, we can go with that.

Reply:
If we fail to create videotestsrc in the if(vtenc) block, we will still exit on the next line below. We can move the check upwards if needed. The downside is that the exit condition is messier.